src/libs/ontuml2gufo/relation_stereotype_functions.ts

```diff
@@ -0,0 +1,381 @@
+      namedNode('rdfs:subPropertyOf'),
+      namedNode('gufo:inheresIn'),
+    ),
+  ];
+}
+
+export function transformComparative(
+  relation: IRelation,
+  options: IOntoUML2GUFOOptions,
+): Quad[] {
+  const { id, name } = relation;
+  const uri = getURI({
+    id,
+    name,
+    uriFormatBy: options.uriFormatBy,
+    relation,
+  });
+
+  return [
+    quad(
+      namedNode(`:${uri}`),
+      namedNode('rdf:type'),
+      namedNode('gufo:ComparativeRelationshipType'),
+    ),
+  ];
+}
+
+export function transformComponentOf(
+  relation: IRelation,
+  options: IOntoUML2GUFOOptions,
+): Quad[] {
+  const { id, name } = relation;
+  const uri = getURI({
+    id,
+    name,
+    uriFormatBy: options.uriFormatBy,
+    relation,
+  });
+
+  return [
+    quad(
+      namedNode(`:${uri}`),
+      namedNode('rdfs:subPropertyOf'),
+      namedNode('gufo:isComponentOf'),
+    ),
+  ];
+}
+
+export function transformCreation(
+  relation: IRelation,
+  options: IOntoUML2GUFOOptions,
+): Quad[] {
+  const { id, name } = relation;
+  const uri = getURI({
+    id,
+    name,
+    uriFormatBy: options.uriFormatBy,
+    relation,
+  });
+
+  return [
+    quad(
+      namedNode(`:${uri}`),
+      namedNode('rdfs:subPropertyOf'),
+      namedNode('gufo:wasCreatedIn'),
+    ),
+  ];
+}
+
+export function transformDerivation(
+  relation: IRelation,
+  options: IOntoUML2GUFOOptions,
+): Quad[] {
+  const { properties } = relation;
+  const domain = properties[0].propertyType.id;
+  const range = properties[1].propertyType.id;
+
+  return [
+    quad(
+      namedNode(`:${domain}`),
+      namedNode('gufo:isDerivedFrom'),
+      namedNode(`:${range}`),
+    ),
+  ];
+}
+
+export function transformExternalDependence(
+  relation: IRelation,
+  options: IOntoUML2GUFOOptions,
+): Quad[] {
+  const { id, name } = relation;
+  const uri = getURI({
+    id,
+    name,
+    uriFormatBy: options.uriFormatBy,
+    relation,
+  });
+
+  return [
+    quad(
+      namedNode(`:${uri}`),
+      namedNode('rdfs:subPropertyOf'),
+      namedNode('gufo:externallyDependsOn'),
+    ),
+  ];
+}
+
+export function transformHistorical(
+  relation: IRelation,
+  options: IOntoUML2GUFOOptions,
+): Quad[] {
+  const { id, name } = relation;
+  const uri = getURI({
+    id,
+    name,
+    uriFormatBy: options.uriFormatBy,
+    relation,
+  });
+
+  return [
+    quad(
+      namedNode(`:${uri}`),
+      namedNode('rdfs:subPropertyOf'),
+      namedNode('gufo:historicallyDependsOn'),
+    ),
+  ];
+}
+
+export function transformHistoricalDependence(
+  relation: IRelation,
+  options: IOntoUML2GUFOOptions,
+): Quad[] {
+  const { id, name } = relation;
+  const uri = getURI({
+    id,
+    name,
+    uriFormatBy: options.uriFormatBy,
+    relation,
+  });
+
+  return [
+    quad(
+      namedNode(`:${uri}`),
+      namedNode('rdfs:subPropertyOf'),
+      namedNode('gufo:historicallyDependsOn'),
+    ),
+  ];
+}
+
+export function transformInstantiation(
+  relation: IRelation,
```

Comment:
When transforming this type of relation, you need to **invert** the association ends defined in OntoUML schema.

The reason is that <<instantiation>> has the higher order type as the target and gufo:categorizes has the higher order type as the domain.

Reply:
Done